externals/tXml.js

```diff
@@ -38,6 +38,39 @@
  * @property {(a: tNode, b: tNode) => boolean} [filter]
  */
 
+/**
+ * Entities that are used in XML
+ */
+export const XML_ENTITIES = {
```

Comment:
These are _not_ [XML Entities](https://www.w3.org/TR/xml/#sec-entity-decl). They are escaped character strings, as per the XML specification's [Syntax](https://www.w3.org/TR/xml/#syntax) section.

Entities are a very different, much more powerful (and dangerous, in the wrong hands) construct.

Reply:
That is incorrect. Please refer to [XML §4.1 [68]](https://www.w3.org/TR/xml/#sec-references).

> Definition: An entity reference refers to the content of a named entity.
> Definition: References to parsed general entities use ampersand (&) and semicolon (;) as delimiters.

A quick Google search also shows that "&amp;" qualifies as an XML entity.

Reply:
The XML specification trumps a Wikipedia entry or a Google search!

I see your point, and the part you reference, and that the phrasing used does include the term `entity reference`; however, those entity references are general in syntax and could include non-predefined entities - for clarity, it's those that I was referring to in a generalised sense.

Since the entity references handled by this change are _only_ the predefined entities, and, importantly, not generalised XML entities, I'd suggest the best reference in the XML specification is to [§4.6 Predefined Entities](https://www.w3.org/TR/xml/#sec-predefined-ent), and possibly that's the best terminology to use in the pull request - I see that you arrived at the same conclusion at https://github.com/Dash-Industry-Forum/dash.js/blob/065acfbc86768ee96f012c100e5be86961547f92/externals/tXml.js#L42 so we're in agreement.

Another consideration worth checking here is if the parser handles generic character references of the form `&#60;` or `&#xABC;` - from my reading of the code, it does not, but that would be good additional functionality to add to the `translateEntities()` function.

Reply:
I believe translating character references is outside the scope of this PR, which primarily concerns translating predefined XML entities. Comparatively, XML entity translation is of much greater importance because `&` is frequently used in DASH manifests (URLs containing query strings).

Reply:
Since character references may be used in place of predefined entities (in fact, the predefined entities dereference to character references), this seems like an odd omission to me, and may need further work in the future, but I'm generally happy with keeping the scope of changes as small as possible.

Reply:
(sighs). Okay, I've added code to translate character references as well.